src/highdicom/sr/value_types.py

```diff
@@ -877,19 +877,16 @@ def __init__(
     @property
     def value(self) -> datetime.time:
         """datetime.time: time"""
-        allowed_formats = [
-            '%H:%M:%S.%f',
-            '%H:%M:%S',
-            '%H:%M',
-            '%H',
-        ]
-        for fmt in allowed_formats:
+        if isinstance(self.Time, TM):
+            value = self.Time
+        else:
             try:
-                dt = datetime.datetime.strptime(self.Time.isoformat(), fmt)
-                return dt.time()
-            except ValueError:
-                continue
-        raise ValueError(f'Could not decode time value "{self.Time}"')
+                value = TM(self.Time)
+            except ValueError as exception:
+                raise ValueError(
+                    f'Could not decode time value "{self.Time}"'
+                ) from exception
+        return value.replace()
```

Comment:
I didn't write this code originally (that was @hackermd ) and I'm not sure why it was written like it is nor why it doesn't work. Your approach seems more logical I have to say. But have you tried it with `pydicom.config.datetime_conversion`?

Reply:
I added `config.datetime_conversion = datetime_conversion` as parameter to the `test_from_dataset` and it looks OK. Looking at the DA, TM, and DT inits they either take a string or a date, time, or datetime.